### PR TITLE
tests: Upgrade psycopg3 version

### PR DESCRIPTION
**Pull request by def-**
To get `https://github.com/psycopg/psycopg/pull/670`

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

**Comment by nrainer-materialize** [comment]
Oh, no:

> [Failed] Test (def-:pr-psycopg3) #92043
tests: Upgrade psycopg3 version - Dennis Felsing (7bf4b00ff517)
[View Build](https://buildkite.com/materialize/test/builds/92043)
Testdrive 5: Uploading results failed! connection failed: connection to server at "18.211.11.249", port 6875 failed: SSL SYSCALL error: EOF detected

(https://materializeinc.slack.com/archives/C05CQUVHY48/p1728553011618789)

**Comment by def-** [comment]
> Oh, no:
> 
> > [Failed] Test (def-:pr-psycopg3) #92043
> > tests: Upgrade psycopg3 version - Dennis Felsing ([7bf4b00](https://github.com/MaterializeInc/materialize/commit/7bf4b00ff517d4da9823b9c2b7b473de2296cd32))
> > [View Build](https://buildkite.com/materialize/test/builds/92043)
> > Testdrive 5: Uploading results failed! connection failed: connection to server at "18.211.11.249", port 6875 failed: SSL SYSCALL error: EOF detected
> 
> (https://materializeinc.slack.com/archives/C05CQUVHY48/p1728553011618789)

I have added keepalives, let's see if that helps.

File: ci/test/pipeline.template.yml [hunk 1]
```diff
@@ -137,7 +137,7 @@ steps:
         inputs:
           - "*"
         depends_on: []
-        timeout_in_minutes: 10
+        timeout_in_minutes: 20
```
Comment (nrainer-materialize) [inline ci/test/pipeline.template.yml:140]: Unrelated (and outdated?) change

Comment (def-) [reply ci/test/pipeline.template.yml:140]: Here this hung for 10 minutes just doing the docker pull, so I think we still want this. (but yes, unrelated)

Comment (nrainer-materialize) [reply ci/test/pipeline.template.yml:140]: Ok

**Comment by nrainer-materialize** [comment]
These debugging issues in output-consistency do no longer occur with 3.2.3: 
```
    return waiting.wait(gen, self.pgconn.socket, timeout=timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "psycopg_binary/_psycopg/waiting.pyx", line 179, in psycopg_binary._psycopg.wait_c
InterruptedError: [Errno 4] Interrupted system call
```

👍

**Pull request merged**